onnx/backend/test/runner/__init__.py

```diff
@@ -169,15 +169,18 @@ def tests(self):  # type: () -> Type[unittest.TestCase]
     def assert_similar_outputs(cls, ref_outputs, outputs, rtol, atol):  # type: (Sequence[Any], Sequence[Any], float, float) -> None
         np.testing.assert_equal(len(outputs), len(ref_outputs))
         for i in range(len(outputs)):
-            np.testing.assert_equal(outputs[i].dtype, ref_outputs[i].dtype)
-            if ref_outputs[i].dtype == np.object:
+            if isinstance(outputs[i], (list, tuple)):
                 np.testing.assert_array_equal(outputs[i], ref_outputs[i])
```

Comment:
I don't understand how this is supposed to work. In particular, I wonder if this assumes some restrictions on the types of the values stored in the list. Will it work for a sequence of sequence of tensors, for example? I think it may be better to recursively call assert_similar_outputs for every pair of corresponding elements.

Reply:
> I don't understand how this is supposed to work. In particular, I wonder if this assumes some restrictions on the types of the values stored in the list. Will it work for a sequence of sequence of tensors, for example? I think it may be better to recursively call assert_similar_outputs for every pair of corresponding elements.

I checked onnx operator output, it only support seq(tensor) . But it's good idea to to recursively call assert_similar_outputs(). I will update. Thanks.

Reply:
@gramalingam I updated to call assert_similar_outputs() recursively, please review.